components/esp_netif/lwip/esp_netif_lwip_slip.c

```diff
@@ -0,0 +1,243 @@
+#include "esp_netif_slip.h"
+
+#include "esp_netif_lwip_internal.h"
+
+#include "lwip/opt.h"
+#include "lwip/sio.h"
+#include "lwip/ip.h"
+#include "lwip/ip6.h"
+#include "lwip/ip6_addr.h"
+#include "lwip/netif.h"
+#include "netif/slipif.h"
+
+#include <string.h>
+
+
+ESP_EVENT_DEFINE_BASE(NETIF_SLIP_STATUS);
+
+static const char *TAG = "esp-netif_lwip-slip";
+
+/**
+ * @brief LWIP SLIP context object
+ */
+typedef struct lwip_slip_ctx {
+    //! ESP netif object
+    esp_netif_t *esp_netif;
+
+    //! SLIP interface IP6 address
+    ip6_addr_t addr;
+
+
+    //! UART device for underling SIO
+    uart_port_t uart_dev;
+
+} lwip_slip_ctx_t;
+
+/**
+ * @brief Create a new lwip slip interface
+ */
+lwip_slip_ctx_t *esp_netif_new_slip(esp_netif_t *esp_netif, const esp_netif_netstack_config_t *esp_netif_stack_config)
+{
+    ESP_LOGD(TAG, "%s", __func__);
+
+    // Fetch netif and create context
+    struct netif *netif_impl = esp_netif->lwip_netif;
+
+    lwip_slip_ctx_t *slip_ctx = calloc(1, sizeof(lwip_slip_ctx_t));
+    if (slip_ctx == NULL) {
+        ESP_LOGE(TAG, "%s: cannot allocate lwip_slip_ctx_t", __func__);
+        return NULL;
+    }
+
+    ESP_LOGD(TAG, "%s: Initialising SLIP (esp_netif %p, lwip_netif %p, device: %d)", __func__, esp_netif, netif_impl, slip_ctx->uart_dev);
+
+    // Load config
+    const esp_netif_slip_config_t *slip_config = &esp_netif_stack_config->lwip_slip.slip_config;
+
+    // Attache network interface and uart device
+    slip_ctx->uart_dev = slip_config->uart_dev;
+    slip_ctx->esp_netif = esp_netif;
+
+    ESP_LOGI(TAG, "%s: Created SLIP interface (netif %p, slip_ctx: %p)", __func__, esp_netif, slip_ctx);
+
+    return slip_ctx;
+}
+
+/**
+ * @brief Creates new SLIP related structure
+ */
+esp_err_t esp_netif_start_slip(lwip_slip_ctx_t *slip_ctx)
+{
+    ESP_LOGI(TAG, "%s: Starting SLIP connection (slip_ctx: %p, addr: %s)", __func__, slip_ctx, ip6addr_ntoa(&slip_ctx->addr));
+
+    struct netif *netif_impl = slip_ctx->esp_netif->lwip_netif;
+
+    // Store serial port number in net interface for SIO open command
+    netif_impl->state = (void *)slip_ctx->uart_dev;
+
+    // Attach interface
+    netif_add_noaddr(slip_ctx->esp_netif->lwip_netif, (void *)slip_ctx->uart_dev, &slipif_init, &ip_input);
+
+    // Bind address
+    int8_t addr_index = 0;
+    //netif_add_ip6_address(slip_ctx->esp_netif->lwip_netif, &slip_ctx->addr, &addr_index);
+
+    netif_ip6_addr_set(slip_ctx->esp_netif->lwip_netif, addr_index, &slip_ctx->addr);
+    netif_ip6_addr_set_state(slip_ctx->esp_netif->lwip_netif, addr_index, IP6_ADDR_VALID);
+
+    // Setup interface
+    netif_set_link_up(slip_ctx->esp_netif->lwip_netif);
+    netif_set_up(slip_ctx->esp_netif->lwip_netif);
+
+    return ESP_OK;
+}
+
+/**
+ * @brief Stops the SLIP interface
+ */
+esp_err_t esp_netif_stop_slip(lwip_slip_ctx_t *slip_ctx)
+{
+    ESP_LOGI(TAG, "%s: Stopped SLIP connection: %p", __func__, slip_ctx);
+
+    // TODO: stop / unbind interface
```

Comment:
Should we be calling the opposite of `netif_set_link_up` and `netif_set_up` here?

Reply:
Yes please, `netif_set_link_down()` seems appropriate.